### PR TITLE
Docs update regarding transforms and timeouts

### PR DESCRIPTION
**Pull request by nnmm**
While writing this, I noticed I couldn't reproduce the problem in https://github.com/tier4/Pilot.Auto/issues/23#issuecomment-708116278, and in that light `lookupTransform()` actually seems superior to `waitForTransform()`. I'll follow up on that and edit this PR accordingly.

See https://docs.google.com/spreadsheets/d/1BvFIMwp0kSkQecw2dkkyj4kR9edFLRDDBBTGynHBAN8/edit?usp=sharing for the gritty details.

### CODE REVIEW

File: docs/porting-to-ROS2.md [hunk 1]
```diff
@@ -362,17 +369,34 @@ There is a [migration guide](https://index.ros.org/doc/ros2/Tutorials/Launch-fil
 ### Replacing `tf2_ros::Buffer`
 A `tf2_ros::Buffer` member that is filled by a `tf2_ros::TransformListener` can become a `tf2::BufferCore` in most cases. This reduces porting effort, since the a `tf2::BufferCore` can be constructed like a ROS1 `tf2_ros::Buffer`. For an example, see [this PR](https://github.com/tier4/Pilot.Auto/pull/11).
 
-However, in some cases the extra functionality of `tf2_ros::Buffer` is needed. For instance, waiting for a transform to arrive.
+However, in some cases the extra functionality of `tf2_ros::Buffer` is needed. For instance, waiting for a transform to arrive, usually in the form of `lookupTransform()` with a timeout argument.
+
+#### Avoiding a loookup with timeout
```
Comment (esteve) [inline docs/porting-to-ROS2.md]: s/loookup/lookup/

File: docs/porting-to-ROS2.md [hunk 2]
```diff
@@ -383,13 +407,24 @@ You could do therefore try setting up a dedicated thread, but you could also use
...
+    } else {
+      // The transform is here, and can now be accessed without triggering the waiting-infinitely bug
+      auto transform = tf_future.get();
+    }
+        
 
-The callback will always be called, but only after some time: when the transform becomes available or when the timeout is reached. In the latter case, if the transform is not ready yet, calling `.get()` on the future will throw a `tf2::TimeoutException`.
+The `waitForTransform()` function will return immediately. Note that the timeout passed to `waitForTransform()` does not matter, only the timeout passed to `wait_for()`.
+
+There is a bug with this when `tf2::TimeStampZero` is requested instead of a nonzero time: the status will be `ready`, but accessing the result with `get()` throws an exception. There is another bug where this bug is not triggered under some conditions, for extra fun. So do not use `waitForTransform()` with `tf2::TimeStampZero` (or any other way of saying "time 0"). The silver lining is that this can be often avoided anyway since it is the scenario described in the section before.
```
Comment (esteve) [inline docs/porting-to-ROS2.md:428]: Is this bug reported somewhere?

File: docs/porting-to-ROS2.md [hunk 3]
```diff
@@ -113,7 +115,12 @@ That's where differences start to show – I decided to make the `VehicleCmdGate
 
 
 ### Latched topics
-For each latched publisher, I just used `transient_local` QoS on the publisher, and we will need to do the same on all subscribers to that topic.
+For each latched publisher, you can use `transient_local` durability QoS on the publisher, e.g. when the history depth is 1:
+
+    rclcpp::QoS durable_qos{1};
```
Comment (esteve) [inline docs/porting-to-ROS2.md:120]: `transient_local()` returns the `rclcpp::QoS` object, so you can synthesize into a one-liner.

**Review by esteve — CHANGES_REQUESTED** [review]

File: docs/porting-to-ROS2.md (hunk 2 above)
Comment (nnmm) [reply docs/porting-to-ROS2.md:428]: I'm planning to do it.

File: docs/porting-to-ROS2.md (hunk 3 above)
Comment (nnmm) [reply docs/porting-to-ROS2.md:120]: True. Should I add it?

Comment (esteve) [reply docs/porting-to-ROS2.md:120]: I think it looks cleaner as a one-liner, but it's up to you.

Comment (nnmm) [reply docs/porting-to-ROS2.md:120]: Done. If you want, feel free to merge.

**Review by esteve — APPROVED** [review]

**Comment by esteve** [comment]
@nnmm thanks for addressing my feedback.

**Pull request merged**